sechub-schedule/src/test/java/com/daimler/sechub/domain/schedule/config/SchedulerProjectConfigServiceTest.java

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: MIT
+package com.daimler.sechub.domain.schedule.config;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import java.util.Optional;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.daimler.sechub.domain.schedule.ScheduleAssertService;
+import com.daimler.sechub.sharedkernel.project.ProjectAccessLevel;
+
+class SchedulerProjectConfigServiceTest {
+
+    private static final String PROJECT1 = "project1";
+    private SchedulerProjectConfigService serviceToTest;
+    private SchedulerProjectConfigRepository repository;
+    private ScheduleAssertService assertService;
+
+    @BeforeEach
+    void beforeEach() {
+        repository = mock(SchedulerProjectConfigRepository.class);
+        assertService = mock(ScheduleAssertService.class);
+
+        serviceToTest = new SchedulerProjectConfigService();
+        serviceToTest.assertService = assertService;
+        serviceToTest.repository = repository;
+
+    }
+
+    @Test
+    void change_to_null_former_full_access_throws_illegal_argument() {
+        assertThrows(IllegalArgumentException.class, () -> serviceToTest.changeProjectAccessLevel(PROJECT1, null, ProjectAccessLevel.READ_ONLY));
+    }
+
+    @Test
+    void change_to_read_only_with_former_null__project_id_is_validated_and_set_to_wanted_access_level_id() {
+        /* prepare */
+        SchedulerProjectConfig config = new SchedulerProjectConfig();
+        config.projectId = PROJECT1;
+        config.projectAccessLevel = ProjectAccessLevel.FULL;
+
+        when(repository.findById(PROJECT1)).thenReturn(Optional.of(config));
+
+        /* execute */
+        serviceToTest.changeProjectAccessLevel(PROJECT1, ProjectAccessLevel.READ_ONLY, null);
+
+        /* test */
+        verify(repository).save(config);
+        verify(assertService).asserProjectIdValid(PROJECT1);
+        assertEquals(ProjectAccessLevel.READ_ONLY, config.getProjectAccessLevel());
+    }
+
+    @Test
+    void change_to_read_only_with_former_full_access__project_id_is_validated_and_set_to_wanted_access_level_id() {
+        /* prepare */
+        SchedulerProjectConfig config = new SchedulerProjectConfig();
+        config.projectId = PROJECT1;
+        config.projectAccessLevel = ProjectAccessLevel.FULL;
+
+        when(repository.findById(PROJECT1)).thenReturn(Optional.of(config));
+
+        /* execute */
+        serviceToTest.changeProjectAccessLevel(PROJECT1, ProjectAccessLevel.READ_ONLY, null);
+
+        /* test */
+        verify(repository).save(config);
+        verify(assertService).asserProjectIdValid(PROJECT1);
```

Comment:
I think, a `t` is missing in the method name `asserProjectIdValid`

```suggestion
        verify(assertService).assertProjectIdValid(PROJECT1);
```

Reply:
done